### DISCUSSION TITLE
Publishing MQTT JSON Does Nothing

### DISCUSSION THREAD

**Discussion by rj45jack** [meshtastic/firmware#3988] [Troubleshooting]
Hi,

I am trying to publish a MQTT JSON for my Meshtastic to read and send out to my private mesh radio. I have dug through several Issues in GH and Posts on Reddit , Discourse and others. I don't think I've found a solution. I have read the documentation regarding MQTT and the JSON, and I came up with the below.

- I am running my own MQTT broker.
- Created a channel called MQTT on my meshtastic.
- Crafted a test payload with the minimum requirements to publish.
- Using MQTT Explorer I publish the Json payload to the topic. msh/2/json/mqtt
- Enabled up/down link on my primary private channel 0.
- Enabled Downlink on mqtt channel.
- I started with the stable firmware, but have moved to the latest alpha for testing.

While monitoring the logs via USB I found the attached error. 

`WARN | 12:25:53 1311 [mqtt] JSON downlink received on channel not called 'mqtt' or without downlink enabled.`

Here is my MQTT Payload/JSON

`{
  "from": "625433568",
  "type": "sendtext",
  "payload": "this is a test"
}`

Are there parameters I am missing in anything there, Payload or Configuration. I feel like I am close to a solution, but I fear there is an issue with MT reading JSON as indicated by several other users in [this thread on Reddit.](https://www.reddit.com/r/meshtastic/comments/1aukc64/anyone_had_succes_sending_meesage_originating/)

**Comment by rj45jack** [comment]
I've expanded my testing to Node-Red. Sending a JSON payload via node-red injection also does not work. But I now appear to be in the correct channel/topic for MQTT. Here is an output from the serial monitor.

`ERROR | 14:16:32 14 [mqtt] JSON Received payload on MQTT but not a valid envelope.`

I am not sure what a valid envelope is, but I am using node-red's and mqtt-explorer JSON functions.

**Comment by rj45jack** [comment]
Well I've given up on using MQTT to push payloads/messages out Meshtastic. I cannot get the correct envelope. I asked on Discord, Github, Reddit, Disourse, I have found zero answers on how to properly get this working.

**Comment by gertsch7003** [comment]
i have done it the other way around. i have sent a message to the mesh and hooked up a mqtt broker on the other end. i have recieved json payload within nodered: 

{"topic":"msh/gw1/2/json/LongFast/!335d7a38","payload":{"channel":0,"from":862597848,"hops_away":0,"id":1534351354,"payload":{"text":"Hallo101"},"rssi":-50,"sender":"!335d7a38","snr":5.75,"timestamp":0,"to":4294967295,"type":"text"},"qos":0,"retain":false,"_topic":"msh/gw1/2/json/LongFast/!335d7a38","_msgid":"c5bd739f49d5329d"}

I have configures a topic inside node red with "msh/gw1/#" (the /# is the major point here )
-> the topic inside meshtastic is msh/gw1

This was the message: Hallo101

hope that will help

Best
